### PR TITLE
EventSource: Convert some htm tests to JS

### PR DESCRIPTION
**Pull request by MierenManz**
These should be able to be safely ported to js without breaking.

I mostly did this so that [Deno](https://github.com/denoland/deno) can run these tests without needing to emulate the DOM

### CODE REVIEW

File: eventsource/eventsource-onopen.any.js
Comment (annevk) [inline eventsource/eventsource-onopen.any.js:1]: This is wrong.

**Review by annevk — COMMENTED** [review]
Can you add a newline after the `// META`?

Instead of `setTimeout` you can use `t.step_timeout` and then remove these from `lint.ignore` (or update the file names in `lint.ignore`, but might as well fix it while here).

**Comment by MierenManz** [comment]
> Can you add a newline after the // META?
Oh yeah sure. Are there other things I need to change aswell?
Other than the `// META: url` typo?

File: eventsource/eventsource-onopen.any.js
Comment (MierenManz) [reply eventsource/eventsource-onopen.any.js]: This should now be fixed

**Comment by MierenManz** [comment]
This is done and ready for review.
I hope there is nothing wrong with the changes.

File: eventsource/eventsource-prototype.any.js
Comment (annevk) [inline eventsource/eventsource-prototype.any.js:7]:
```suggestion
        assert_own_property(self, "EventSource")
```
If you want this to run in workers (or rename the file).

File: eventsource/eventsource-request-cancellation.any.js
Comment (annevk) [inline eventsource/eventsource-request-cancellation.any.js:9]: I think `t.step_timeout` here and below would be clearer. At least it's not clear to me that `this` is equivalent to `t`.

File: eventsource/format-field-id-2.any.js
Comment (annevk) [inline eventsource/format-field-id-2.any.js:1]: This is wrong and this wouldn't even run. Did you try running these?

**Review by MierenManz — COMMENTED** [review]
I removed a duplicate test from the suite.
The only difference was testing `document.documentURI` and `self.location.toString()`
@andreubotella had suggested this to me and it looks fine to me

File: eventsource/request-cache-control.any.js
Comment (MierenManz) [inline eventsource/request-cache-control.any.js:9]: This file seems to get stuck on a timeout.
@andreubotella can probably explain this change better than I can.

Comment (andreubotella) [reply eventsource/request-cache-control.any.js:9]: Each of the top-level calls to `cacheTest` created two async tests, one at the moment of calling the function with `runAgain` set to `true`, and once again when that test succeeded. However, since the second test for each URL was created asynchronously, they might be created after the `load` event fires, and so the test might have completed by that time. This seemed to mostly work on the test as it was, but changing the script from an inline script to a fetch seemed to break this.

This change instead creates all async tests before the `load` event.

**Review by andreubotella — APPROVED** [review]
Other than the three test files which should be window-only tests, this LGTM.

**Comment by andreubotella** [comment]
`eventsource-constructor-empty-url` and `eventsource-constructor-url-bogus` worked fine as `.any.js` tests, since they didn't use any window-specific APIs and were specifically marked as supporting the window and worker globals. I'd leave them as they were.

**Review by andreubotella — APPROVED** [review]
LGTM but I'll wait for @annevk's review before merging.

File: eventsource/format-utf-8.any.js
Comment (annevk) [inline eventsource/format-utf-8.any.js:12]:
```suggestion
})

```

**Review by annevk — APPROVED** [review]
Apart from a tiny nit this looks okay to me, though I scanned it rather quickly this time.

**Comment by MierenManz** [comment]
commited the suggested change.

**Comment by annevk** [comment]
Sigh, I noticed a problem.

`eventsource-constructor-url-bogus.any.js` was renamed to `eventsource-constructor-url-bogus.js`.

Can you restore that?

**Comment by annevk** [comment]
I think next time we do one of these we should probably do it in small increments if it's not mechanical as it's too easy to miss things like that.

**Comment by MierenManz** [comment]
> Sigh, I noticed a problem.
> 
> `eventsource-constructor-url-bogus.any.js` was renamed to `eventsource-constructor-url-bogus.js`.
> 
> Can you restore that?

Should I open a quick pr for that?

**Comment by annevk** [comment]
Yeah please!

**Pull request merged**
